sdks/wasm/framework/src/WebAssembly.Net.Http/WasmHttpMessageHandler.cs

```diff
@@ -364,30 +362,40 @@ public override async Task<int> ReadAsync (byte [] buffer, int offset, int count
 						return 0;
 					}
 
-					using (var body = _status.Body) {
-						_reader = (JSObject)body.Invoke ("getReader");
+					try { 
+						using (var body = _status.Body) {
+							_reader = (JSObject)body.Invoke ("getReader");
+						}
+					} catch (JSException) {
+						cancellationToken.ThrowIfCancellationRequested ();
+						throw;
 					}
 				}
 
 				if (_bufferedBytes != null && _position < _bufferedBytes.Length) {
 					return ReadBuffered ();
 				}
 
-				var t = (Task<object>)_reader.Invoke ("read");
-				using (var read = (JSObject)await t) {
-					if ((bool)read.GetObjectProperty ("done")) {
-						_reader.Dispose ();
-						_reader = null;
+				try {
+					var t = (Task<object>)_reader.Invoke ("read");
+					using (var read = (JSObject)await t) {
+						if ((bool)read.GetObjectProperty ("done")) {
```

Comment:
Maybe check if passed cancellation token is cancelled after awaiting `read`? (Not required for this fix / PR.)